### PR TITLE
Linter for global function names

### PR DESCRIPTION
**Pull request by rambleraptor**
Issue: https://github.com/kristerkari/stylelint-scss/issues/397

This creates a linter for global function names.

Couple things:

1) I didn't know how comprehensive to make the docs + test suite. Putting in examples for every rule seemed excessively verbose.

2) I purposefully didn't put in auto-fix functionality, since we're encouraging people to use the migrator. Does that seem safe?

### CODE REVIEW

**Comment by rambleraptor** [comment]
@nex3

File: src/rules/no-global-function-names/__tests__/index.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+testRule(rule, {
+  ruleName,
+  config: [true],
+  syntax: "scss",
+
+  accept: [
+    {
+      code: `
+      @use color;
```
Comment (nex3) [inline src/rules/no-global-function-names/__tests__/index.js]:
```suggestion
      @use "sass:color";
```

Also below.

File: src/rules/no-global-function-names/__tests__/index.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      a {
+       background: color.red(#6b717f);
+      }
+    `,
+      description: "Non-global function with different name"
+    }
+  ],
```
Comment (nex3) [inline src/rules/no-global-function-names/__tests__/index.js]: I'd also add accept tests for using a global function that's still allowed (such as `rgb()` and/or `if()`).

**Comment by rambleraptor** [comment]
changes made!

**Comment by kristerkari** [comment]
Looks good!

**Comment by kristerkari** [comment]
Hmm.. maybe we should generate test cases for each function? If we don't test them then the test coverage will suffer as there will be so many lines that are not tested.

**Comment by rambleraptor** [comment]
I had some spare time and put in a whole bunch more test cases. There's more that need to be put in, but this is a lot more than we had earlier.

**Comment by kristerkari** [comment]
I think that this is good now. I need to look through the changes once more before merging.

**Pull request merged**